xarray/core/variable.py

```diff
@@ -1099,66 +1099,85 @@ def shift(self, shifts=None, fill_value=dtypes.NA, **shifts_kwargs):
+        mode: str = "constant",
+        stat_length: Union[
+            int, Tuple[int, int], Mapping[Hashable, Tuple[int, int]]
+        ] = None,
+        constant_values: Union[
+            int, Tuple[int, int], Mapping[Hashable, Tuple[int, int]]
+        ] = None,
+        end_values: Union[
+            int, Tuple[int, int], Mapping[Hashable, Tuple[int, int]]
+        ] = None,
+        reflect_type: str = None,
+        **pad_widths_kwargs: Any,
     ):
         """
-        Return a new Variable with paddings.
+        Return a new Variable with padded data.
 
         Parameters
         ----------
-        pad_width: Mapping of the form {dim: (before, after)}
-            Number of values padded to the edges of each dimension.
-        **pad_widths_kwargs:
-            Keyword argument for pad_widths
+        pad_widths: Mapping with the form of {dim: (pad_before, pad_after)}
+            Number of values padded along each dimension.
+        mode: (str)
+            See numpy / Dask docs
+        **pad_widths_kwarg:
+            One of pad_widths or pad_widths_kwarg must be provided.
+
+        Returns
+        -------
+        padded : Variable
+            Variable with the same dimensions and attributes but padded data.
         """
         pad_widths = either_dict_or_kwargs(pad_widths, pad_widths_kwargs, "pad")
 
-        if fill_value is dtypes.NA:
-            dtype, fill_value = dtypes.maybe_promote(self.dtype)
+        # change default behaviour of pad with mode constant
+        if mode == "constant" and constant_values is None:
+            dtype, constant_values = dtypes.maybe_promote(self.dtype)
         else:
             dtype = self.dtype
 
-        if isinstance(self.data, dask_array_type):
-            array = self.data
+        # create pad_options_kwargs, numpy requires only relevant kwargs to be nonempty
+        if isinstance(stat_length, dict):
+            stat_length = [
+                (n, n) if d not in stat_length else stat_length[d]
+                for d, n in zip(self.dims, self.data.shape)
+            ]
+        if isinstance(constant_values, dict):
+            constant_values = [
+                (0, 0) if d not in constant_values else constant_values[d]
+                for d, n in zip(self.dims, self.data.shape)
+            ]
+        if isinstance(end_values, dict):
+            end_values = [
+                (0, 0) if d not in end_values else end_values[d]
+                for d, n in zip(self.dims, self.data.shape)
+            ]
+
+        # workaround for bug in Dask's default value of stat_length  https://github.com/dask/dask/issues/5303
+        if stat_length is None and mode in ["maximum", "mean", "median", "minimum"]:
+            stat_length = [(n, n) for n in self.data.shape]
+
+        pads = [(0, 0) if d not in pad_widths else pad_widths[d] for d in self.dims]
+
+        # numpy/dask work with optional kwargs
+        pad_option_kwargs = {}
+        if stat_length is not None:
```

Comment:
are these if conditions needed? just pass these kwargs directly in the `duck_array_ops.pad` call below even if they are None?

Reply:
Unfortunately, yes.